### PR TITLE
feat: DocString result names for Safe-DS stub results

### LINKED ISSUE #100: Use result names in docstrings for result names in generated stubs
### Is your feature request related to a problem?

Results currently always have names `result1`, `result2`, etc. Frequently, however, the docstrings contain proper names for the results.

### Desired solution

Use result names in docstrings for result names in generated stubs.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Masara**
Closes #100

### Summary of Changes

Result names for stubs from Numpy docstring are now supported. Google and ReST don't support multiple results or named results, therefore they handle these a bit different:
* For Google docstrings, if there are multiple results we only get the first result docstring (type & description).
* For ReST we get all types, but only the last result description in the docstring.

### CODE REVIEW

File: src/safeds_stubgen/stubs_generator/_generate_stubs.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class NamingConvention(IntEnum):
     SAFE_DS = 2
 
 
-def generate_stubs(api: API, out_path: Path, convert_identifiers: bool) -> None:
+def generate_stubs(api: API, out_path: Path, convert_identifiers: bool, is_numpy_parser: bool = False) -> None:
```
Comment (lars-reimann) [inline src/safeds_stubgen/stubs_generator/_generate_stubs.py:31]: This should work regardless of `is_numpy_parser`. Any necessary preprocessing must have been done while creating the `API`. That's the reason we have this intermediate representation.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds_stubgen/stubs_generator/_generate_stubs.py (hunk 1 above)
Comment (Masara) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: I removed the check and some unnecessary fields for the Result type.

**Comment by Masara** [comment]
@lars-reimann The linter failed to run
```
Run actions/upload-artifact@v4
Multiple search paths detected. Calculating the least common ancestor of all paths
The least common ancestor is /home/runner/work/Stub-Generator/Stub-Generator. This will be the root directory of the artifact
With the provided path, there will be 10 files uploaded
Artifact name is valid!
Root directory input is valid!
Beginning upload of artifact content to blob storage
Uploaded bytes 7[16](https://github.com/Safe-DS/Stub-Generator/actions/runs/8636183463/job/23675494721?pr=101#step:5:17)2
Finished uploading artifact content to blob storage!
SHA256 hash of uploaded artifact zip is eb6b45b691dee50ca3d7608b93b[18](https://github.com/Safe-DS/Stub-Generator/actions/runs/8636183463/job/23675494721?pr=101#step:5:19)b3b85f0a8ad228204896204a503d9caebe8
Finalizing artifact upload
Error: Failed to FinalizeArtifact: Received non-retryable error: Failed request: (403) Forbidden: Error from intermediary with HTTP status code 403 "Forbidden"
```

**Comment by Masara** [comment]
Nevermind, it runs now

**Comment by Masara** [comment]
@lars-reimann Could you merge this with higher priority? This is required for issue #80

File: src/safeds_stubgen/docstring_parsing/_docstring.py [hunk 2]
```diff
@@ -47,10 +48,13 @@ def to_dict(self) -> dict[str, Any]:
         return dataclasses.asdict(self)
 
 
+@dataclass(frozen=True)
+class ResultDocstrings:
+    docstrings: Sequence[ResultDocstring]
+
+
```
Comment (lars-reimann) [inline src/safeds_stubgen/docstring_parsing/_docstring.py:55]: What is the purpose of this class, as opposed to passing a list of `ResultDocstring` around?

Comment (Masara) [reply src/safeds_stubgen/docstring_parsing/_docstring.py]: That's... a good question. I removed it.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**